src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -97,4 +97,22 @@ else if(ledmode == LEDMode.BLINK){
     }
 
   }
+  
+  public Pose2d shiftRight(){
+    if(hasTarget()){
+      double[] arr = botpose.getDoubleArray(new double[6]);
+      return new Pose2d(arr[0], arr[1], Rotation2d.fromDegrees(arr[5]));
+    }else{
+      return null;
+    }
+  }
+
+  public Pose2d shiftLeft(){
+    if(hasTarget()){
+      double[] arr = botpose.getDoubleArray(new double[6]);
+      return new Pose2d(arr[0], arr[1], Rotation2d.fromDegrees(arr[5]));
+    }else{
+      return null;
+    }
+  }
```

Comment:
I think these methods are not needed as they are the same as our existing pose getting method. 

Also our default drive command is already set up to continually check for the target and to update the pose if one is seen. 

I think this means that you can focus most of your effort on the translation / shifting part of your goal